vendor/bat-native-ledger/src/bat/ledger/internal/ledger_impl.cc

```diff
@@ -187,23 +155,30 @@ void LedgerImpl::InitializeDatabase(
   database()->Initialize(execute_create_script, database_callback);
 }
 
-void LedgerImpl::OnInitialized(
-    const type::Result result,
-    ledger::ResultCallback callback) {
-  initializing_ = false;
+void LedgerImpl::OnInitialized(type::Result result, ResultCallback callback) {
+  DCHECK(ready_state_ == ReadyState::kInitializing);
 
   if (result == type::Result::LEDGER_OK) {
     StartServices();
   } else {
     BLOG(0, "Failed to initialize wallet " << result);
   }
 
+  while (!ready_callbacks_.empty()) {
```

Comment:
I would maybe use a `std::queue<>` instead, as it would make the intent more clear.